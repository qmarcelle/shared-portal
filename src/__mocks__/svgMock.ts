import React from 'react';

const SvgMock = React.forwardRef<
  HTMLSpanElement,
  React.SVGProps<SVGSVGElement>
>((props, ref) => {
  return React.createElement('span', {
    ref,
    ...props,
    'data-testid': 'svg-mock',
  });
});

SvgMock.displayName = 'SvgMock';

export default SvgMock;
export const ReactComponent = SvgMock;
