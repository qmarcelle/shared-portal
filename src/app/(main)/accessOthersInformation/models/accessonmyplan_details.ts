export interface AccessOnMyPlanDetails {
  memberName: string;
  DOB: string;
  isOnline: boolean;
}
