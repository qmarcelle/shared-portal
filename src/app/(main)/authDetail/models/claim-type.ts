export enum ClaimType {
  Medical = 'Medical',
  Pharmacy = 'Pharmacy',
  Dental = 'Dental',
}
