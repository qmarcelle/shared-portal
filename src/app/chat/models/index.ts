// Export all chat-related types
export type {
  CobrowseConfig,
  CobrowseInitResponse,
  CobrowseSession,
  CobrowseSessionResponse,
} from './cobrowse';

// Make sure the ChatMessage type is consistent by defining it inline here
export interface ChatMessage {
  id: string;
  text: string;
  sender: 'user' | 'agent' | 'bot';
  timestamp: Date;
}

export * from './plans';
export type {
  ChatEligibility,
  ChatPayload,
  ChatSession,
  ChatSessionJWT,
  ChatState,
  PlanSwitcherState,
} from './session';
export type {
  BusinessDay,
  BusinessHours,
  ChatConfig,
  ChatOption,
  CobrowseState,
  UserEligibility,
} from './types';
