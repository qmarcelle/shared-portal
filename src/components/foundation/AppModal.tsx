import FocusTrap from 'focus-trap-react';
import Image from 'next/image';
import React, { ReactElement, createContext } from 'react';
import { Modal } from 'react-responsive-modal';
import { create } from 'zustand';
import leftIcon from '../../../public/assets/left.svg';
import { IComponent } from '../IComponent';
import { Column } from './Column';
import { bcbstSilhouletteLogo, closeIcon } from './Icons';
import { Row } from './Row';
import { Spacer } from './Spacer';
import { TextBox } from './TextBox';

/* eslint-disable @typescript-eslint/no-explicit-any */
interface ModalProps extends IComponent {
  content: ReactElement & ModalChildProps;
  store?: any;
  isChildActionAppModal?: boolean;
  childModalContent?: ReactElement & ModalChildProps;
}
interface ModalControllerProps {
  showBack: boolean;
  showModal: boolean;
  pageIndex: number;
  store: null;
  isChildAction?: boolean;
  updateShowBack: (isShowBack: boolean) => void;
  updateshowModal: (isShowBack: boolean) => void;
  updatepageIndex: (pageIndex: number) => void;
  updateStore: (store: null) => void;
  showAppModal: (props: ModalProps) => void;
  dismissModal: () => void;
  isChildModal: boolean;
  showChildModal: () => void;
  dismissChildModal: () => void;
  isChildActionAppModalProp: boolean;
}

export const useAppModalStore = create<ModalControllerProps>((set) => ({
  showBack: false,
  showModal: false,
  isChildModal: false,
  pageIndex: 0,
  store: null,
  isChildActionAppModalProp: false,
  updateShowBack: (isShowBack: boolean) =>
    set(() => ({ showBack: isShowBack })),
  updateshowModal: (isShowModal: boolean) =>
    set(() => ({ showModal: isShowModal })),
  updatepageIndex: (pageIndex: number) => set(() => ({ pageIndex: pageIndex })),
  updateStore: (store: null) => set(() => ({ store: store })),
  showAppModal: ({
    content,
    store,
    isChildActionAppModal,
    childModalContent,
  }: ModalProps) => {
    modalBody = content;
    childModalBody = childModalContent;
    if (isChildActionAppModal) {
      set(() => ({
        isChildActionAppModalProp: isChildActionAppModal,
      }));
    }
    set(() => ({
      showModal: true,
    }));
    set(() => ({
      store: store,
    }));
    pageIndexStack = [0];
  },
  dismissModal: () => {
    set(() => ({
      showModal: false,
    }));
    set(() => ({
      store: null,
    }));
    set(() => ({
      showBack: false,
    }));
    set(() => ({
      pageIndex: 0,
    }));
    set(() => ({
      isChildModal: false,
    }));
    modalBody = null;
  },
  showChildModal: () => {
    set(() => ({
      isChildModal: true,
    }));
  },
  dismissChildModal: () => {
    set(() => ({
      isChildModal: false,
    }));
  },
}));
let modalBody: ReactElement | null = null;
let childModalBody: ReactElement | null | undefined = null;
let pageIndexStack: number[] = [];
export interface ModalChildProps {
  changePage?: (pageIndex: number, showBackButton?: boolean) => any;
  pageIndex?: number;
}

export const ModalContext = createContext<any>(null);

interface ModalHeaderProps {
  onClose: () => any;
}

const ModalHeader = ({ onClose }: ModalHeaderProps) => {
  const { updatepageIndex, updateShowBack, pageIndex, showBack } =
    useAppModalStore();
  const onBackPressed = () => {
    pageIndexStack.pop();
    const popIndex = pageIndexStack[pageIndexStack.length - 1];
    updatepageIndex(popIndex ?? 0);
    if (pageIndex == 0) {
      updateShowBack(false);
    }
  };

  const handleKeyDown = (e: {
    preventDefault: () => void;
    keyCode: number;
  }) => {
    e.preventDefault();
    if (e.keyCode === 13) {
      onClose();
    }
  };

  return (
    <Row className="justify-between modal-header items-center relative">
      {showBack ? (
        <Row
          tabIndex={1}
          onClick={onBackPressed}
          className="items-center focus-icon"
        >
          <Image src={leftIcon} className="modal-icon" alt="back" />
          <TextBox text="Back" className="primary-color underline" />
        </Row>
      ) : (
        <div tabIndex={1} />
      )}

      <Image
        src={bcbstSilhouletteLogo}
        className="modal-icon modal-header-logo absolute m-auto left-0 right-0"
        alt="bcbst logo"
      />
      <div
        tabIndex={0}
        className="size-8 focus-icon"
        onKeyDown={handleKeyDown}
        onClick={onClose}
      >
        <Image src={closeIcon} className="size-4" alt="close" />
      </div>
    </Row>
  );
};

const ModalFooter = () => {
  return (
    <Row className="justify-center modal-footer">
      <Image
        src={bcbstSilhouletteLogo}
        className="modal-icon"
        alt="bcbst logo"
      />
    </Row>
  );
};

export const AppModal = () => {
  const {
    dismissModal,
    updateShowBack,
    updatepageIndex,
    store,
    showModal,
    pageIndex,
    isChildModal,
    showChildModal,
    dismissChildModal,
    isChildActionAppModalProp,
  } = useAppModalStore();
  const changePage = (pageIndex: number, showBackButton: boolean = false) => {
    updateShowBack(showBackButton);
    updatepageIndex(pageIndex);
    pageIndexStack.push(pageIndex);
  };
  const closeModal = () => {
    isChildActionAppModalProp ? showChildModal() : dismissModal();
  };

  const closeChildModal = () => {
    dismissChildModal();
  };
  const closeAllModal = () => {
    dismissModal();
  };
  return (
    <ModalContext.Provider value={store}>
      <InnerAppModal
        showModal={showModal}
        modalBody={modalBody ?? <></>}
        changePage={changePage}
        pageIndex={pageIndex}
        closeModal={closeModal}
        isChildModal={isChildModal}
        closeChildModal={closeChildModal}
        closeAllModal={closeAllModal}
        childModalBody={childModalBody ?? <></>}
      />
    </ModalContext.Provider>
  );
};

type InnerAppModalProps = {
  showModal?: boolean;
  closeModal?: () => any;
  modalBody: ReactElement;
  changePage?: (pageIndex: number, showBackButton: boolean) => any;
  pageIndex?: number;
  isChildModal?: boolean;
  closeChildModal?: () => any;
  closeAllModal?: () => any;
  childModalBody?: ReactElement;
};

export const InnerAppModal = ({
  showModal = true,
  closeModal = () => {},
  modalBody,
  changePage = () => {},
  pageIndex = 0,
  isChildModal = false,
  childModalBody,
}: InnerAppModalProps) => {
  return (
    <Modal
      classNames={{
        overlay: 'modal-overlay',
      }}
      showCloseIcon={false}
      open={showModal}
      closeOnOverlayClick={true}
      onClose={closeModal}
    >
      <FocusTrap>
        <div>
          <Column className="modal-container">
            <ModalHeader onClose={closeModal} />
            <Spacer size={32} />
            <div className="flex flex-row flex-grow justify-center">
              <div className="flex flex-row justify-center modal-content">
                {modalBody ? (
                  React.cloneElement(modalBody, {
                    changePage,
                    pageIndex: pageIndex,
                  })
                ) : (
                  <div />
                )}
              </div>
            </div>
            {isChildModal && (
              <div className="child-modal">
                <div className="child-modal-content">{childModalBody}</div>
              </div>
            )}
            <Spacer size={32} />
            <ModalFooter />
          </Column>
        </div>
      </FocusTrap>
    </Modal>
  );
};
