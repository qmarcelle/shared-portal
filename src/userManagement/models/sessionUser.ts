import { VisibilityRules } from '@/visibilityEngine/rules';

export enum UserRole {
  MEMBER = 'MEM',
  PERSONAL_REP = 'PR',
  AUTHORIZED_USER = 'AU',
  NON_MEM = 'NM',
}

export type SessionUser = {
  id: string;
  currUsr: {
    umpi: string;
    fhirId: string;
    role: UserRole;
    plan:
      | {
          memCk: string;
          sbsbCk: string;
          grgrCk: string;
          grpId: string;
          subId: string;
          fhirId: string;
          ntwkId: string;
        }
      | undefined;
  };
  rules?: string;
  vRules?: VisibilityRules;
  subscriberId?: string;
};
