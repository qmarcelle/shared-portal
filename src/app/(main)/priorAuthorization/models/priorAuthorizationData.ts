import { ClaimDetails } from '../../../../models/claim_details';

export const priorAuthorizationData: ClaimDetails[] = [
  {
    id: 'Claim1',
    claimStatus: 'Pending',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Physical Therapy',
    memberName: 'Chris Hall',
    serviceDate: '04/06/23',
    totalBilled: '',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'Anand Patel',
    ReferredTo: 'Anand Patel',
    priorAuthFlag: true,
  },
  {
    id: 'Claim2',
    claimStatus: 'Approved',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Magnetic Resonance Imaging (MRI)',
    memberName: 'Chris Hall',
    serviceDate: '01/23/23',
    totalBilled: '145.00',
    planPaid: '114.76',
    myShare: '30.24',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'N/A',
    ReferredTo: 'N/A',
    priorAuthFlag: true,
  },
  {
    id: 'Claim3',
    claimStatus: 'Partial Approval',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Continuous Positive Airway Pressure Machiâ€¦',
    memberName: 'Chris Hall',
    serviceDate: '08/10/23',
    totalBilled: '403.98',
    planPaid: '338.37',
    myShare: '65.61',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'N/A',
    ReferredTo: 'Anand Patel',
    priorAuthFlag: true,
  },
  {
    id: 'Claim4',
    claimStatus: 'Denied',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Home Health Services',
    memberName: 'Chris Hall',
    serviceDate: '07/05/23',
    totalBilled: '403.98',
    planPaid: '0.00',
    myShare: '403.98',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'Home Health Services',
    ReferredTo: 'N/A',
    priorAuthFlag: true,
  },
  {
    id: 'Claim5',
    claimStatus: 'Approved',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Brittant Johnson',
    memberName: 'Chris Hall',
    serviceDate: '07/04/23',
    totalBilled: '183.00',
    planPaid: '150.85',
    myShare: '32.15',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'N/A',
    ReferredTo: 'N/A',
    priorAuthFlag: true,
  },
];
