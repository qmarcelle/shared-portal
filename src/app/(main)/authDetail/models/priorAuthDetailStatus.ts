import { ClaimStatus } from '@/app/(main)/authDetail/models/claim-status';
import { ClaimType } from '@/app/(main)/authDetail/models/claim-type';

export interface PriorAuthDetailsStatus {
  priorAuthDetailType: ClaimType;
  priorAuthDetailName: string;
  dateOfVisit: string;
  priorAuthDetailStatus: ClaimStatus;
  member: string;
  PriorAuthReferenceId: string;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  authInfo: any;
  referredName: string;
}
