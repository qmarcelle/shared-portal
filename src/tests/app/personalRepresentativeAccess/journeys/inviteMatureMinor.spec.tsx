import PersonalRepresentativePage from '@/app/personalRepresentativeAccess/page';
import { AppModal, useAppModalStore } from '@/components/foundation/AppModal';
import { mockedAxios } from '@/tests/__mocks__/axios';
import { mockedFetch } from '@/tests/setup';
import { fetchRespWrapper } from '@/tests/test_utils';
import { UserRole } from '@/userManagement/models/sessionUser';
import '@testing-library/jest-dom';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';

jest.mock('src/auth', () => ({
  auth: jest.fn(() =>
    Promise.resolve({
      user: {
        id: 'testUser',
        currUsr: {
          umpi: '65d4458442db105635427958',
          role: UserRole.PERSONAL_REP,
          plan: {
            planName: 'BlueCross BlueShield of Tennessee',
            subId: '123456',
            grpId: '100000',
            memCk: '',
            coverageType: ['Medical', 'Dental', 'Vision'],
          },
        },
        vRules: {
          matureMinor: false,
        },
      },
    }),
  ),
}));

jest.mock('src/utils/date_formatter', () => ({
  formatDateToLocale: jest.fn(),
}));

const renderUI = async () => {
  const component = await PersonalRepresentativePage();
  const { container } = render(
    <>
      <AppModal />
      {component}
    </>,
  );

  return container;
};
describe('Inviting Mature Minor to regsiter online', () => {
  beforeEach(() => {
    const mockFormatDate = jest.requireMock(
      'src/utils/date_formatter',
    ).formatDateToLocale;
    mockFormatDate.mockReturnValueOnce('01/18/1985');
    const { dismissModal } = useAppModalStore.getState();
    dismissModal();
  });
  it('should show member details and Invite', async () => {
    mockedFetch
      .mockResolvedValueOnce(
        fetchRespWrapper({
          data: {
            getPBEmessage: 'Person Record fetched Successfully ',
            getConsentmessage: 'Person Record fetched Successfully ',
            getPBECorrelationId: '0124b43b-efd0-42bc-b904-cb2ac9c712b9',
            consentCorrelationId: 'cd30f2b3-3bae-42f3-b93a-8fa42f3884e7',
            getPBEDetails: [
              {
                umpid: '57c85test3ebd23c7db88245',
                userName: 'Testuser553',
                personFHIRID: '30345928-abcd-ef01-2345-6789abcdef51',
                firstName: 'Quino',
                middleName: 'S',
                lastName: 'Deper',
                suffix: 'Mr.',
                address1: 'street1',
                address2: 'street2',
                phoneNumber: '8095469997',
                city: 'Ongole',
                state: 'AP',
                zip: 34566,
                dob: '2019-10-29T00:00:10.0000000+00:00',
                gender: 'M',
                email: 'sub.m@gmail.com',
                relationshipInfo: [
                  {
                    personRoleType: 'PR',
                    org: 'bcbst_facets',
                    roleTermDate: '2030-11-30T00:00:00.0000000+00:00',
                    nativeId: '38922455201-100000',
                    primaryPlanFlag: false,
                    patientFHIRID: '30345928-abcd-ef01-2345-6789abcdef52',
                    userName: 'Testuser553',
                    memeCk: '3349134',
                    clientId: '194',
                    multiPlanConfirmed: false,
                    multiPlanConfirmedDate: '2030-11-30T00:00:00.0000000+00:00',
                    approvalRequestId: '38922455201-1123456787',
                    relatedPersons: [
                      {
                        relatedPersonUMPID: '57c85test3ebd23c7db88244',
                        relatedPersonFirstName: 'Rafusal',
                        relatedPersonLastName: 'Claud',
                        relatedPersonMiddleName: 'S',
                        relatedPersonSuffix: 'Mr.',
                        relatedPersonNativeId: '38922455200-100000',
                        relatedPersonFHIRID:
                          'ddd94652-d077-454d-b252-bcb7c24e1de5',
                        relatedPersonPatientFHIRID: '',
                        relatedPersonRelationshipTermDate:
                          '2030-11-30T00:00:00.0000000+00:00',
                        relatedPersonRoleType: 'Subscriber',
                        relatedPersonDob: '2030-11-30T00:00:00.0000000+00:00',
                        relatedPersonApprovalRequestId: '',
                        relatedPersonMemeCk: '502622001',
                      },
                      {
                        relatedPersonUMPID: '57c85test3ebd23c7db88244',
                        relatedPersonFirstName: 'Rafusal',
                        relatedPersonLastName: 'Claud',
                        relatedPersonMiddleName: 'S',
                        relatedPersonSuffix: 'Mr.',
                        relatedPersonNativeId: '38922455200-100000',
                        relatedPersonFHIRID:
                          'ddd94652-d077-454d-b252-bcb7c24e1de5',
                        relatedPersonPatientFHIRID: '',
                        relatedPersonRelationshipTermDate:
                          '2030-11-30T00:00:00.0000000+00:00',
                        relatedPersonRoleType: 'Subscriber',
                        relatedPersonDob: '2030-11-30T00:00:00.0000000+00:00',
                        relatedPersonApprovalRequestId: '',
                        relatedPersonMemeCk: '846239401',
                      },
                    ],
                  },
                ],
              },
            ],
            consentDetails: [
              {
                id: 'e7d4e7e9-5f98-4e35-9141-de8c8eb944a6',
                name: 'Consent - Limited Access',
                description:
                  'This consent is governed by limited access policy for HIV and Cancer information',
                status: 'active',
                performer: 'john.doe@bcbst.com',
                requester: 'Person/24baffdb-76b7-4cb1-bf97-47f4bcf5574a',
                requestee: 'Patient/e3c212e5-1b7f-4521-8ba1-3b48befa98d6',
                policyId: 'ce3ee5b4-3810-4c8a-bfd9-9d8681853a7e',
                policyBusinessIdentifier: 'pol_10237_lim_ac',
                type: 'PEOPLE_TO_PEOPLE',
                effectiveOn: '2023-01-01T12:00:00Z',
                expiresOn: '2023-12-31T12:00:00Z',
                signedOn: '2023-01-01T12:00:00Z',
                modifiedAt: '2023-01-01T12:00:00Z',
                accessControl: 'PERMIT_ALL',
                options: [
                  {
                    id: 'f83a70bf-736d-4dfa-96eb-31d5c5a09535',
                    businessIdentifier: 'option_biz_id_001',
                    caption: 'HIV',
                    mappings: [
                      {
                        criteria: 'CODE',
                        key: 'http://hl7.org/fhir/sid/icd-10',
                        value: 'B20,B9735',
                      },
                    ],
                  },
                ],
              },
            ],
          },
          details: {
            componentName: 'string',
            componentStatus: 'string',
            returnCode: 'string',
            subSystemName: 'string',
            message: 'string',
            problemTypes: 'string',
            innerDetails: {
              statusDetails: [
                {
                  componentName: 'string',
                  componentStatus: 'string',
                  returnCode: 'string',
                  subSystemName: 'string',
                  message: 'string',
                  problemTypes: 'string',
                  innerDetails: {
                    statusDetails: ['string'],
                  },
                },
              ],
            },
          },
        }),
      )
      .mockResolvedValueOnce(
        fetchRespWrapper({
          data: {
            getPBEmessage: 'Person Record fetched Successfully ',
            getConsentmessage: 'Person Record fetched Successfully ',
            getPBECorrelationId: '0124b43b-efd0-42bc-b904-cb2ac9c712b9',
            consentCorrelationId: 'cd30f2b3-3bae-42f3-b93a-8fa42f3884e7',
            getPBEDetails: [
              {
                umpid: '57c85test3ebd23c7db88245',
                userName: 'Testuser553',
                personFHIRID: '30345928-abcd-ef01-2345-6789abcdef51',
                firstName: 'Rafusal',
                middleName: 'S',
                lastName: 'Claud',
                suffix: 'Mrs.',
                address1: 'street1',
                address2: 'street2',
                phoneNumber: '8095469997',
                city: 'Ongole',
                state: 'AP',
                zip: 34566,
                dob: '2019-10-29T00:00:10.0000000+00:00',
                gender: 'M',
                email: 'sub.m@gmail.com',
                hasAccount: false,
                relationshipInfo: [
                  {
                    personRoleType: 'PR',
                    org: 'bcbst_facets',
                    roleTermDate: '2030-11-30T00:00:00.0000000+00:00',
                    nativeId: '38922455201-100000',
                    primaryPlanFlag: false,
                    patientFHIRID: '30345928-abcd-ef01-2345-6789abcdef52',
                    userName: 'Testuser553',
                    memeCk: '3349134',
                    clientId: '194',
                    multiPlanConfirmed: false,
                    multiPlanConfirmedDate: '2030-11-30T00:00:00.0000000+00:00',
                    approvalRequestId: '38922455201-1123456787',
                    relatedPersons: [
                      {
                        relatedPersonUMPID: '57c85test3ebd23c7db88244',
                        relatedPersonFirstName: 'Rafusal',
                        relatedPersonLastName: 'Claud',
                        relatedPersonMiddleName: 'S',
                        relatedPersonSuffix: 'Mr.',
                        relatedPersonNativeId: '38922455200-100000',
                        relatedPersonFHIRID:
                          'ddd94652-d077-454d-b252-bcb7c24e1de5',
                        relatedPersonPatientFHIRID: '',
                        relatedPersonRelationshipTermDate:
                          '2030-11-30T00:00:00.0000000+00:00',
                        relatedPersonRoleType: 'Subscriber',
                        relatedPersonDob: '2030-11-30T00:00:00.0000000+00:00',
                        relatedPersonApprovalRequestId: '',
                        relatedPersonMemeCk: '502622001',
                        name: 'Full Access',
                      },
                      {
                        relatedPersonUMPID: '57c85test3ebd23c7db88244',
                        relatedPersonFirstName: 'Rafusal',
                        relatedPersonLastName: 'Claud',
                        relatedPersonMiddleName: 'S',
                        relatedPersonSuffix: 'Mr.',
                        relatedPersonNativeId: '38922455200-100000',
                        relatedPersonFHIRID:
                          'ddd94652-d077-454d-b252-bcb7c24e1de5',
                        relatedPersonPatientFHIRID: '',
                        relatedPersonRelationshipTermDate:
                          '2030-11-30T00:00:00.0000000+00:00',
                        relatedPersonRoleType: 'Subscriber',
                        relatedPersonDob: '2030-11-30T00:00:00.0000000+00:00',
                        relatedPersonApprovalRequestId: '',
                        relatedPersonMemeCk: '846239401',
                        name: 'Full Access',
                      },
                    ],
                  },
                ],
              },
            ],
            consentDetails: [
              {
                id: 'e7d4e7e9-5f98-4e35-9141-de8c8eb944a6',
                name: 'Consent - Limited Access',
                description:
                  'This consent is governed by limited access policy for HIV and Cancer information',
                status: 'active',
                performer: 'john.doe@bcbst.com',
                requester: 'Person/24baffdb-76b7-4cb1-bf97-47f4bcf5574a',
                requestee: 'Patient/e3c212e5-1b7f-4521-8ba1-3b48befa98d6',
                policyId: 'ce3ee5b4-3810-4c8a-bfd9-9d8681853a7e',
                policyBusinessIdentifier: 'pol_10237_lim_ac',
                type: 'PEOPLE_TO_PEOPLE',
                effectiveOn: '2023-01-01T12:00:00Z',
                expiresOn: '2023-12-31T12:00:00Z',
                signedOn: '2023-01-01T12:00:00Z',
                modifiedAt: '2023-01-01T12:00:00Z',
                accessControl: 'PERMIT_ALL',
                options: [
                  {
                    id: 'f83a70bf-736d-4dfa-96eb-31d5c5a09535',
                    businessIdentifier: 'option_biz_id_001',
                    caption: 'HIV',
                    mappings: [
                      {
                        criteria: 'CODE',
                        key: 'http://hl7.org/fhir/sid/icd-10',
                        value: 'B20,B9735',
                      },
                    ],
                  },
                ],
              },
            ],
          },
          details: {
            componentName: 'string',
            componentStatus: 'string',
            returnCode: 'string',
            subSystemName: 'string',
            message: 'string',
            problemTypes: 'string',
            innerDetails: {
              statusDetails: [
                {
                  componentName: 'string',
                  componentStatus: 'string',
                  returnCode: 'string',
                  subSystemName: 'string',
                  message: 'string',
                  problemTypes: 'string',
                  innerDetails: {
                    statusDetails: ['string'],
                  },
                },
              ],
            },
          },
        }),
      );
    mockedAxios.get.mockResolvedValueOnce({
      data: {
        data: {
          isEmailSent: 'true',
        },
      },
    });

    const component = await renderUI();

    // eslint-disable-next-line prettier/prettier
    expect(screen.getByText('Understanding Access')).toBeVisible();
    expect(screen.getByText('Members You Represent')).toBeVisible();
    expect(screen.getByText('Rafusal Claud')).toBeVisible();
    fireEvent.click(screen.getByText('Invite to Register'));
    expect(
      screen.getByRole('heading', { name: 'Invite to Register' }),
    ).toBeVisible();

    const inputEmail = screen.getAllByRole('textbox', {
      name: /Their Email Address/i,
    })[0];
    const inputConfirmationEmail = screen.getByRole('textbox', {
      name: /Confirm Their Email Address/i,
    });
    fireEvent.change(inputEmail, { target: { value: 'testuser@bcbst.com' } });
    fireEvent.change(inputConfirmationEmail, {
      target: { value: 'testuser@bcbst.com' },
    });
    fireEvent.click(screen.getByRole('button', { name: /Send Invite/i }));
    await waitFor(() => {
      expect(mockedAxios.get).toHaveBeenCalledWith(
        '/userRegistration/sharePermission/sendInvite?memeck=502622001&requesteeFHRID=ddd94652-d077-454d-b252-bcb7c24e1de5&requesteeEmailID=testuser@bcbst.com&requestType=Invite',
      );
    });
    await waitFor(() => {
      expect(
        screen.getByRole('heading', { name: 'Invitation to Register Sent' }),
      ).toBeVisible();
    });
    fireEvent.click(screen.getByRole('button', { name: /Done/i }));
    expect(screen.getByText('Pending...')).toBeVisible();
    expect(component).toMatchSnapshot();
  });
});
