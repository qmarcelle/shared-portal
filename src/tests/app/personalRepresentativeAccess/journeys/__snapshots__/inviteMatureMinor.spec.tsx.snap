// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inviting Mature Minor to regsiter online should show member details and Invite 1`] = `
<div>
  <div
    class="flex flex-col justify-center items-center page"
  >
    <div
      class="flex flex-col app-content app-base-font-color"
    >
      <span
        style="display: block; width: 1px; min-width: 1px; height: 52px; min-height: 52px;"
      />
      <h1
        class="title-1 pl-3"
      >
        Personal Representative Access
      </h1>
      <span
        style="display: block; width: 1px; min-width: 1px; height: 12px; min-height: 12px;"
      />
      <p
        class="body-1 w-2/3 ml-4"
        style="display: block;"
      >
        Personal representatives have the legal authority to make health care decisions on behalf of the member.
      </p>
      <span
        style="display: block; width: 1px; min-width: 1px; height: 22px; min-height: 22px;"
      />
      <section
        class="flex flex-row items-start app-body"
      >
        <div
          class="flex flex-col flex-grow page-section-36_67 items-stretch"
        >
          <div
            class="card-main  large-section"
          >
            <div
              class="flex flex-col items-stretch"
            >
              <div>
                <h1
                  class="title-1 title-2"
                >
                  Understanding Access
                </h1>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 22px; min-height: 22px;"
                />
              </div>
              <div
                class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-4 "
              >
                <div
                  class="flex flex-row m-1 accordion-head"
                >
                  <p
                    class="mr-2 flex-grow"
                  >
                    Full Access
                  </p>
                  <img
                    alt="Down Chevron"
                    class="pl-2 w-6"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                </div>
                <div
                  class="mr-2 flex-grow"
                />
              </div>
              <div
                class=""
                style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
              />
              <div
                class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-4 "
              >
                <div
                  class="flex flex-row m-1 accordion-head"
                >
                  <p
                    class="mr-2 flex-grow"
                  >
                    Basic Access
                  </p>
                  <img
                    alt="Down Chevron"
                    class="pl-2 w-6"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                </div>
                <div
                  class="mr-2 flex-grow"
                />
              </div>
            </div>
          </div>
          <div />
        </div>
        <div
          class="flex flex-col page-section-63_33 items-stretch"
        >
          <div
            class="card-main large-section"
          >
            <div>
              <div
                class="flex flex-col flex flex-col"
              >
                <h2
                  class="title-2"
                >
                  Members You Represent
                </h2>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
                />
                <div
                  class="flex flex-col flex flex-col"
                >
                  <div
                    class="card-elevated cursor-pointer mb-4"
                  >
                    <div
                      class="flex flex-col m-8"
                    >
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                      />
                      <div
                        class="flex flex-row justify-between"
                      >
                        <p
                          class="body-1 font-bold body-1"
                          style="display: block;"
                        >
                          Rafusal Claud
                        </p>
                        <p
                          class="body-1"
                          style="display: block;"
                        >
                          DOB: 01/18/1985
                        </p>
                      </div>
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                      />
                      <div
                        class=""
                        style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                      />
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                      />
                      <div
                        class="flex flex-col"
                      >
                        <div
                          class="flex flex-row"
                        >
                          <p
                            class="body-1 pt-1 ml-1"
                            style="display: block;"
                          >
                            Basic Access as of undefined
                          </p>
                          <span
                            style="display: block; width: 32px; min-width: 32px; height: 1px; min-height: 1px;"
                          />
                        </div>
                        <span
                          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                        />
                        <div
                          class="card-main w-full"
                          style="background-color: rgba(0, 0, 0, 0.05);"
                        >
                          <div
                            class="flex flex-col m-4"
                          >
                            <div
                              class="flex flex-row mt-2"
                            >
                              <p
                                class="body-1 body-1"
                                style="display: block;"
                              >
                                This member has not created an online profile.
                              </p>
                            </div>
                            <div
                              class="flex flex-row"
                            >
                              <div
                                class="mr-2"
                              >
                                <img
                                  alt="link"
                                  data-nimg="1"
                                  decoding="async"
                                  height="40"
                                  loading="lazy"
                                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                                  style="color: transparent;"
                                  width="40"
                                />
                              </div>
                              <p
                                class="mb-4"
                              >
                                Pending...
                              </p>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 12px; min-height: 12px;"
                />
                <div
                  class=""
                  style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                />
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
                />
                <span
                  class="rich-text body-2 undefined"
                >
                  <span>
                    if you have submitted a Personal Representative Authorization form, it might not yet appear on this page. For assistance, please
                  </span>
                  <span
                    class="link"
                  >
                    <a>
                       start a chat 
                    </a>
                  </span>
                  <span>
                    or call us at [1-800-000-000]
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div />
        </div>
      </section>
    </div>
  </div>
</div>
`;
